File: __leetcode.com/Easy/2-7-Reverse-Integer/go/main.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Println(reverse(123))
	fmt.Println(reverse(-456))
	fmt.Println(reverse(2147483641))
	fmt.Println(reverse(2147483648))
	fmt.Println(reverse(-2147483648))
}

func reverse(x int) int {
	res := 0
	if x < 0 {
		res, _ = strconv.Atoi(reverseStr(strconv.Itoa(x)[1:]))
		if -res < -2147483648 {
			return 0
		}
		return -res
	}
	res, _ = strconv.Atoi(reverseStr(strconv.Itoa(x)))
	if res > 2147483648 {
		return 0
	}
	return res
}

func reverseStr(s string) string {
	runes := []rune(s)
	for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
		runes[i], runes[j] = runes[j], runes[i]
	}
	return string(runes)
}

File: __leetcode.com/Easy/2-7-Reverse-Integer/python/main.py
def reverse(x: int) -> int:
    res = 0
    if x < 0:
        res = -int(str(str(x)[1:])[::-1])
        if res < -2147483648:
            return 0
    else:
        res = int(str(x)[::-1])
        if res > 2147483648:
            return 0
    return res


print(reverse(123))
print(reverse(-123))
print(reverse(120))
print(reverse(0))
print(reverse(1534236469))
